core/creators/themedui.js

```diff
@@ -384,6 +388,71 @@ CKEDITOR.replaceClass = 'ckeditor';
 		return editor;
 	}
 
+	// Delay editor creation if given element is detached from DOM.
+	//
+	// Require `config.delayDetached` to be `true`.
+	//
+	// If `config.registerCallback` is declared as function, it will be invoked with single argument:
+	// function, that should be called to create editor.
+	//
+	// Otherwise: periodically (with `setInterval`) run check if element is attached to DOM and create editor automatically.
+	//
+	// ```js
+	//	CKEDITOR.replace( detachedEditorElement, {
+	//		delayDetached: true,
+	//		delayDetachedFrequency: 50,
+	//		registerCallback:RegisterCallback
+	//		}
+	//	} );
+	// ```
+	//
+	// @param {CKEDITOR.element} element The DOM element.
+	// @param {Object} [config] The specific configuration to apply to this
+	// editor instance. Configuration set here will override the global CKEditor settings.
+	// @param {String} [data] Initial value for the instance.
+	// @param {Number} mode CKEditor mode.
+	function delayCreationOnDetachedElement( element, config, data, mode ) {
+		if ( !config || !config.delayDetached || !element.isDetached() ) {
+			return false;
+		}
+
+		var callback = CKEDITOR.config.registerCallback;
```

Comment:
I don't recommend assigning a value to a variable and overwriting it later as it is unclear if that is intended behaviour or mistake.

BTW, here we could have an early return and call the payload instead:

```js
if ( !config.registerCallback ) {
	CKEDITOR.config.registerCallback( payload );
	return true;
}
```

with a second path if config was provided without any overwriting, but it's much easier to spot when you move the declarations after the logic (see the comments concerning `payload` and `registerPayload`).

Note that theoretically we could even ditch the reference to config here and just have `payload()`, but since this config should have a default value anyway, we can as well use it here I guess 🤔